lib/data/network/interceptor/download_file_interceptor.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:dio/dio.dart';
+import 'package:fluffychat/data/network/exception/dio_duplicate_request_exception.dart';
+import 'package:matrix/matrix.dart';
+
+class DownloadFileInterceptor extends InterceptorsWrapper {
+  DownloadFileInterceptor();
+
+  final _currentDownloads = <String>{};
+
+  static const downloadPath = '_matrix/media/v3/download/';
+
+  @override
+  void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
+    Logs().d('DownloadFileInterceptor::onRequest: ${options.path}');
+    if (options.path.contains(downloadPath) &&
+        _currentDownloads.contains(options.path)) {
+      handler.reject(
+        DioDuplicateRequestException(),
+      );
+      return;
+    }
```

Comment:
can you separate this into a method, and write test?

Reply:
okay let me try